sftp/iam.tf

```diff
@@ -101,7 +110,73 @@ data "aws_iam_policy_document" "transfer_server_to_cloudwatch_assume_policy" {
 }
 
 resource "aws_iam_role_policy" "transfer_server_to_cloudwatch_policy" {
+  provider      = aws.ephemeral
   name   = "${var.transfer_server_name}-transfer_server_to_cloudwatch_policy"
   role   = aws_iam_role.transfer_server_role.name
   policy = data.aws_iam_policy_document.transfer_server_to_cloudwatch_assume_policy.json
 }
+
+# Replication policy for S3 to S3 replication
+resource "aws_iam_role" "replication" {
+  provider      = aws.ephemeral
+  name = "sftp-replication-role"
+
+  assume_role_policy = <<POLICY
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "s3.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+POLICY
+}
+
+resource "aws_iam_policy" "replication" {
+  provider      = aws.ephemeral
+  name = "sftp-replication-policy"
+
+  policy = <<POLICY
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": [
+        "s3:GetReplicationConfiguration",
+        "s3:ListBucket"
+      ],
+      "Effect": "Allow",
+      "Resource": [
+        "${aws_s3_bucket.transfer_server_bucket.arn}"
+      ]
+    },
+    {
+      "Action": [
+        "s3:GetObjectVersionForReplication",
+        "s3:GetObjectVersionAcl",
+         "s3:GetObjectVersionTagging"
+      ],
+      "Effect": "Allow",
+      "Resource": [
+        "${aws_s3_bucket.transfer_server_bucket.arn}/*"
+      ]
+    },
+    {
+      "Action": [
+        "s3:ReplicateObject",
+        "s3:ReplicateDelete",
```

Comment:
If we replicate delete, how does this count as 'permanent storage'? Wouldn't deletions in the source/ephemeral bucket propagate to the target/permanent bucket?